force-app/main/default/classes/IBMWatsonIAMOptions.cls

```diff
@@ -1,11 +1,21 @@
 /**
  * Options for authenticating using IAM.
  */
-public class IBMWatsonIAMOptions {
+public class IBMWatsonIAMOptions implements IBMWatsonAuthenticatorConfig {
   private String apiKey;
   private String accessToken;
   private String url;
 
+  public String authenticationType() {
+    return IBMWatsonCredentialUtils.AUTHTYPE_IAM;
+  }
+
+  public void validate() {
+    if (String.isBlank(this.apiKey) && String.isBlank(this.accessToken)) {
+      throw new IBMWatsonServiceExceptions.IllegalArgumentException('You must provide either the apiKey or accessToken properties.');
```

Comment:
Should we check for bad characters in apikey? Were we doing this before? `IBMWatsonCredentialUtils.hasBadStartOrEndChar()`